### PR TITLE
Fix images on auto in download app example page

### LINKED ISSUE #2243: [Download app example page] Images on auto
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

On https://boosted.netlify.app/docs/5.3/examples/download-app/ 

When using 'Light mode', you can see light images:
screenshot-boosted netlify app-2023 09 14-14_59_01

When using 'Dark mode', you can see dark images:
screenshot-boosted netlify app-2023 09 14-14_59_23

When using 'Auto mode', you can see both:
screenshot-boosted netlify app-2023 09 14-14_59_41

I am using Version 117.0.5938.63 (Build officiel) (64 bits)
But I have the same visual issue with Firefox Version 102.14.0esr (64 bits):

firefox

### Reduced test cases

Go to https://boosted.netlify.app/docs/5.3/examples/download-app/  use theme switcher.

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome, Firefox

### What version of Boosted are you using?

v5.3.1 (current main branch)

### PR DESCRIPTION
**Pull request by MewenLeHo**
### Related issues

Closes #2243 

### Description

Following the issue explained in #2243:

(1) I made some tests and it appears that the bug described in the issue only happens on a specific occasion: when your computer is set to 'Light mode' and you use the 'Auto color mode' on the page.

(2) If your your computer is set to 'Dark mode' and you use the 'Auto color mode' on the page the it works perfectly fine, probably because 'Dark mode' is enabled by default.

=> I made a fix in CSS for (1)
=> I also made an additional piece of code (see comment) for (2) as defensive CSS.

### Motivation & Context

Prevent having both light and dark images in  'Auto color mode'.

### Types of change

- Bug fix (non-breaking which fixes an issue)

### Live previews

- <https://deploy-preview-2261--boosted.netlify.app/docs/5.3/examples/download-app/>

#### Contribution

- [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

#### Accessibility

- (na) My change follows accessibility good practices; I have at least run axe

#### Design

- (na) My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
- [x] My change is compatible with a responsive display

#### Development

- [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Developer-guide)
- (na) I have added JavaScript unit tests to cover my changes
- (na) I have added SCSS unit tests to cover my changes

#### Documentation

- (na) My change introduces changes to the documentation and/or I have updated the documentation accordingly

### Checklist (for Core Team only)

- (na) My change introduces changes to the migration guide
- (na) My new component is well displayed in [Storybook](https://deploy-preview-2261--boosted.netlify.app/storybook)
- (na)My new component is compatible with RTL
- [x] Manually run [BrowserStack tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/browserstack.yml)
- [x] Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android)
- [ ] Code review
- (na) Design review
- (na) A11y review

#### After the merge

- [ ] Manually launch [Percy tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/percy.yml)

### CODE REVIEW

File: site/content/docs/5.3/examples/download-app/download-app.css [hunk 1]
```diff
@@ -74,6 +74,18 @@
   display: none;
 }
 
+@media (prefers-color-scheme: light) {
+  [data-bs-theme="auto"] .dark-show {
+    display: none;
+  }
+}
+
+/* @media (prefers-color-scheme: dark) {
```
Comment (MewenLeHo) [inline site/content/docs/5.3/examples/download-app/download-app.css:83]: Fix is working fine without this part.
I just can't decide if we should include it or not as defensive CSS 🤔

**Review by louismaximepiton — APPROVED** [review]
I'm fine landing it as a patch for a bug for v5.3.3 which will arrive soon. I think we don't need any defensive CSS on this.

However, maybe we can fix the color-mode javascript on Bootstrap directly because right now we have:

| `color-mode` on Bootstrap | `color-mode` on device | Result in HTML |
| :--------------------------: | :-----------------------: | :--------------: |
| `dark` | `*` | `data-bs-theme="dark"` |
| `light` | `*` | `data-bs-theme="light"` |
| `auto` | `dark` | `data-bs-theme="dark"` |
| `auto` | `light` | `data-bs-theme="auto"` |

For instance, I was able to reproduce the behavior you patch here by patching `color-mode.js`.

Here is the diff:

```diff
  const setTheme = theme => {
- if (theme === 'auto' && window.matchMedia('(prefers-color-scheme: dark)').matches) {
-   document.documentElement.setAttribute('data-bs-theme', 'dark')
+ if (theme === 'auto') {
+   document.documentElement.setAttribute('data-bs-theme', (window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light'))
  } else {
```

**Comment by MewenLeHo** [comment]
Defensive CSS was removed as requested.

A PR has been opened on Bootstrap's side to update `color-form-js`.

The plan here is to keep the CSS fix presented in the current PR to allow the example page to work correctly.
->it the js fix is not ok for Bootsrap: keep our CSS fix.
->it the js fix is ok for Bootsrap: keep our CSS fix until the js fix is merged in Boosted repository.

**Review by julien-deramond — APPROVED** [review]
LGTM, adding something in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2223 to track it

**Pull request merged**